test/curator.sh

```diff
@@ -0,0 +1,379 @@
+    fi
+    rm -f $origconfig
+    os::log::debug "$( oc set env dc/logging-curator CURATOR_SCRIPT_LOG_LEVEL=INFO CURATOR_LOG_LEVEL=ERROR )"
+    os::log::debug "$( oc set env dc/logging-curator-ops CURATOR_SCRIPT_LOG_LEVEL=INFO CURATOR_LOG_LEVEL=ERROR )"
+    restart_curator errors
+    # this will call declare_test_end, suite_end, etc.
+    os::test::junit::reconcile_output
+}
+trap "cleanup" INT TERM EXIT
+
+os::log::debug "$( oc set env dc/logging-curator CURATOR_SCRIPT_LOG_LEVEL=DEBUG CURATOR_LOG_LEVEL=DEBUG )"
+os::log::debug "$( oc set env dc/logging-curator-ops CURATOR_SCRIPT_LOG_LEVEL=DEBUG CURATOR_LOG_LEVEL=DEBUG || : )"
+fpod=$( oc get pods --selector component=fluentd  -o jsonpath='{ .items[*].metadata.name }' | head -1 )
+tf='%Y.%m.%d'
+today=`date -u +"$tf"`
+yesterday=`date -u +"$tf" --date=yesterday`
+lastweek=`date -u +"$tf" --date="last week"`
+fourweeksago=`date -u +"$tf" --date="4 weeks ago"`
+thirtyonedaysago=`date -u +"$tf" --date="31 days ago"`
+# projects:
+# project-dev-YYYY.mm.dd delete 24 hours
+# project-qe-YYYY.mm.dd delete 7 days
+# project-prod-YYYY.mm.dd delete 4 weeks
+# .operations-YYYY.mm.dd delete 2 months
+
+# where "yesterday", "today", etc. are the dates in UTC in YYYY.mm.dd format
+# When is a month not a month?  When it is a curator month!  When you use
+# month based trimming, curator starts counting at the first day of the
+# current month, not the current day of the current month.  For example, if
+# today is April 15, and you want to delete indices that are 2 months older
+# than today (--time-unit months --older-than 2 --timestring %Y.%m.%d),
+# curator doesn't delete indices that are dated older than February 15, it
+# deletes indices older than _February 1_.  That is, it goes back to the
+# first day of the current month, _then_ goes back two whole months from
+# that date.
+# If you want to be exact with curator, it is best to use
+# day based trimming e.g. --time-unit days --older-than 60 --timestring %Y.%m.%d
+# There is apparently no way to tell curator to delete indices exactly N
+# months older than the current date.
+twomonthsago=`date -u +"$tf" --date="$(date +%Y-%m-1) -2 months -1 day"`
+# project-dev-yesterday project-dev-today
+# project-qe-lastweek project-qe-today
+# project-prod-4weeksago project-today
+# .operations-2monthsago .operations-today
+
+TEST_DIVIDER="------------------------------------------"
+
+test_project_name_errors() {
+    curpod=`get_running_pod curator`
+    curtest=`mktemp --suffix=.yaml`
+    cat > $curtest <<EOF
+this-project-name-is-far-far-too-long-this-project-name-is-far-far-too-long-this-project-name-is-far-far-too-long-this-project-name-is-far-far-too-long:
+  delete:
+    days: 1
+EOF
+    update_config_and_restart $curtest errors
+    sleep 1
+    rc=0
+    # curator pod should be in error state
+    errpod=`get_error_pod curator`
+    if ! os::cmd::expect_success "test -n '$errpod'" ; then
```

Comment:
These are also cases where the `if` doesn't make sense -- let `os::cmd ` handle the failure if you want the failure to fail the job